### ISSUE TITLE
Insert parameters names not supported by Oracle

### ISSUE THREAD

**Issue by RaiTamarindo** [typeorm/typeorm#1512]
**Issue type:**

 [ ] question
 [x] bug report
 [ ] feature request
 [ ] documentation issue

**Database system/driver:**

 [ ] `cordova`
 [ ] `mongodb`
 [ ] `mssql`
 [ ] `mysql` / `mariadb`
 [x] `oracle`
 [ ] `postgres`
 [ ] `sqlite`
 [ ] `sqljs`
 [ ] `websql`

**TypeORM version:**

 [ ] `latest`
 [x] `@next`
 [ ] `0.x.x` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**

The insert query generated by the typeorm has the prefix `_inserted_`, but Oracle does not accept the `_` character in the first position of the parameter identifier.

Example:

Database:
```sql
CREATE TABLE MYSCHEMA.MY_TABLE (
  ID INTEGER   NOT NULL ,
  NAME VARCHAR(32)   NOT NULL,
  DESCRIPTION VARCHAR(256),
PRIMARY KEY(ID) );

CREATE SEQUENCE MYSCHEMA.MY_TABLE_ID_SEQ START WITH 1;

CREATE TRIGGER MYSCHEMA.MY_TABLE_ID_AUTOINC 
BEFORE INSERT ON MYSCHEMA.MY_TABLE
FOR EACH ROW
BEGIN
  :NEW.ID := MYSCHEMA.MY_TABLE_ID_SEQ.NEXTVAL;
END;
/
```

JSON schema:
```json
{
    "name": "Table",
    "table":
    {
        "name": "TABLE"
    },
    "columns":
    {
        "id":
        {
            "name": "ID",
            "primary": true,
            "type": "int",
            "generated": true
        },
        "name":
        {
            "name": "NAME",
            "type": "varchar2"
        },
        "description":
        {
            "name": "DESCRIPTION",
            "type": "varchar2"
        }
    }
}
```

Trying to save:
```javascript
const typeorm = require('typeorm')

typeorm
    .createConnection(
    {
        type: 'oracle',
        host: '127.0.0.1',
        port: 1521,
        sid: 'mydatabase',
        username: 'user',
        password: 'password',
        entitySchemas:
        [
            __dirname + '/../entities/*.schema.json'
        ],
        schema: 'MYSCHEMA',
        entityPrefix: 'MY_'
    })
    .then(function(conn)
    {
        conn.getRepository("Table")
            .save(
            {
                name: 'Name Test',
                description: 'Testing'
            })
            .then(function(result)
            {
                console.log(result);
            })
            .catch(function(err)
            {
                console.log(err);
            });
    });
```

I'm getting the error `ORA-01036: illegal variable name / number` from Oracle, and when I inspect `err.query`, I get the query:
```
INSERT INTO "MYSCHEMA"."MY_TABLE"("NAME", "DESCRIPTION") VALUES (:_inserted_0_NAME, :_inserted_0_DESCRIPTION)
```

If I run this query in SQL Developer, I get the error `ORA-00911: invalid character` and the cause:

>identifiers may not start with any ASCII character other than letters and numbers.  $#_ are also allowed after the first character.  Identifiers enclosed by doublequotes may contain any character other than a doublequote.  Alternative quotes (q'#...#') cannot use spaces, tabs, or carriage returns as delimiters.  For all other contexts, consult the SQL Language Reference Manual.

So I did the following test:

I changed the `InsertQueryBuilder.js` file where I had` var paramName = "_inserted_" + `I replaced with` var paramName = "inserted_" + `and tried to save again.

This time I got no error, but the `save` promise is never resolved. When I checked the table, I saw that the insert had been held.

**Comment by pleerock** [comment]
Currently there are lot of issue with oracle and its reworked and soon will be published in `next` version.

**Comment by AlexMesser** [comment]
fixed and released in `typeorm@0.2.0-alpha.23`. You can try it via `npm i typeorm@next`.
